lib/ClangImporter/ImportType.cpp

```diff
@@ -924,24 +924,29 @@ namespace {
             SmallVector<clang::ObjCProtocolDecl *, 4> protocols{
               type->qual_begin(), type->qual_end()
             };
-            auto *nsObjectProto =
-                Impl.getNSObjectProtocolType()->getAnyNominal();
-            if (!nsObjectProto) {
-              // Input is malformed
-              return {};
+
+            // Skip if there is no NSObject protocol.
+            auto nsObjectProtoType =
+                Impl.getNSObjectProtocolType();
+            if (nsObjectProtoType) {
```

Comment:
What happens in the `else` case?

Reply:
It will be treated like the other non NSObject<...> types, except if it hits another special case it is imported as a protocol composition of the class and the protocols.

Reply:
Oh, right, so far we've just tested that "we have protocols" and "the class is NSObject". Okay.